uui/components/pickers/DataPickerFooter.tsx

```diff
@@ -53,7 +55,7 @@ function DataPickerFooterImpl<TItem, TId>(props: PropsWithChildren<DataPickerFoo
             <FlexSpacer />
 
             <FlexCell width="auto" alignSelf="center">
-                {view.selectAll && (
+                {view.selectAll && !isEmptyRowsAndHasNoSelection && (
```

Comment:
Let's better disable this button, rather than hide it. You can do it via `isDisabled` prop of `LinkButton`